src/pvecontrol/sanitycheck.py

```diff
@@ -0,0 +1,196 @@
+from pvecontrol.utils import fonts
+
+class CheckType(Enum):
+  HA = 'HIGHT_AVAILABILITY'
+  Node = "NODE"
+
+class CheckCode(Enum):
+  CRIT = 'CRITICAL'
+  WARN = 'WARNING'
+  INFO = 'INFO'
+  OK = 'OK'
+
+ICONS = {
+  CheckCode.CRIT.value: '❌',
+  CheckCode.WARN.value: '⚠️',
+  CheckCode.INFO.value: 'ℹ️',
+  CheckCode.OK.value: '✅',
+}
+
+class CheckMessage:
+  def __init__(self, code: CheckCode, message):
+    self.code = code
+    self.message = message
+
+  def display(self, padding_max_size):
+    padding = padding_max_size - len(self.message)
+    print(f"{self.message}{padding * '.'}{ICONS[self.code.value]}")
+
+  def __len__(self):
+    return len(self.message)
+
+class Check:
+
+  def __init__(self, type: CheckType, name: str, messages = None):
+    if messages is None:
+      messages = []
+    self.type = type
+    self.name = name
+    self.messages = messages
+
+  def add_messages(self, messages):
+    if isinstance(messages, CheckMessage):
+      self.messages.append(messages)
+    elif isinstance(messages, list):
+      self.messages += messages
+
+  def set_code(self, code: CheckCode):
+    self.code = code
+
+  def display(self, padding_max_size):
+    print(f"{fonts.BOLD}{self.name}{fonts.END}\n")
+    for msg in self.messages:
+      msg.display(padding_max_size)
+    print()
+
+class SanityCheck():
+
+    check_methods = (
+      "_validate_nodes_capacity",
+      "_validate_ha_groups",
+      "_validate_ha_vms",
+    )
+
+    check_display_order = [
+      CheckType.HA,
+      CheckType.Node,
+    ]
+
+    def __init__(self, proxmox: PVECluster):
+      self._proxmox = proxmox
+      self._ha = self._proxmox.ha()
+      self._checks = []
+
+    def run(self):
+      for method in SanityCheck.check_methods:
+        self._checks.append(getattr(self, method)())
+
+    def _get_longest_message(self):
+      size = 0
+      for check in self._checks:
+        for msg in check.messages:
+          if len(msg) > size:
+            size = len(msg)
+      return size + 1
+
+    def display(self):
+      size = self._get_longest_message()
+      current_type = None
+      for check in self._checks:
+        if current_type != check.type:
+          current_type = check.type
+          dash_size = int((size - (len(check.type.value) + 2))/2)
+          print(f"{dash_size*'-'} {check.type.value} {dash_size*'-'}\n")
+        check.display(size)
+
+    def _check_cpu_overcommit(self, maxcpu, cpufactor, allocated_cpu):
+      return (maxcpu *  cpufactor) <= allocated_cpu
+
+    def _check_mem_overcommit(self, max_mem, min_mem, allocated_mem):
+      return (allocated_mem +  min_mem) >= max_mem
+
+    # Check nodes capacity
+    def _validate_nodes_capacity(self):
+      node_config = self._proxmox.config['node']
+      check = Check(CheckType.Node, "Check Node capacity")
+      for node in self._proxmox.nodes:
+        if self._check_cpu_overcommit(node.maxcpu, node_config['cpufactor'], node.allocatedcpu):
```

Comment:
Not sure if it's normal, but it seems that `cpufactor` is not always present in `node_config`:

```
[pyenv:.env][17:27:24.91][pvecontrol][feat-add-sanitychecks-ha-vms *=][1]:~/dev/internal/pvecontrol$ pvecontrol -c preprod sanitycheck
INFO:root:Proxmox cluster: preprod
Traceback (most recent call last):
  File "/home/paul/dev/internal/pvecontrol/.env/bin/pvecontrol", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/paul/dev/internal/pvecontrol/src/pvecontrol/__init__.py", line 129, in main
    args.func(proxmoxcluster, args)
  File "/home/paul/dev/internal/pvecontrol/src/pvecontrol/actions/cluster.py", line 57, in action_sanitycheck
    sc.run()
  File "/home/paul/dev/internal/pvecontrol/src/pvecontrol/sanitycheck.py", line 82, in run
    self._checks.append(getattr(self, method)())
                        ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/paul/dev/internal/pvecontrol/src/pvecontrol/sanitycheck.py", line 113, in _validate_nodes_capacity
    if self._check_cpu_overcommit(node.maxcpu, node_config['cpufactor'], node.allocatedcpu):
                                               ~~~~~~~~~~~^^^^^^^^^^^^^
KeyError: 'cpufactor'
```

(If I print `node_config` I get an empty dict)

Reply:
Need to be rebase i will did it.